src/Loggers/LogEmbeddings.jl

```diff
@@ -0,0 +1,58 @@
+"""
+
+"""
+function log_embeddings(logger::TBLogger, name::AbstractString, mat::AbstractArray; metadata=nothing, labels=nothing, metadata_header=nothing, step=nothing)
+    @assert ndims(mat) == 2 "Embedding matrix must be 2-D"
+    matrix_path = joinpath(logger.logdir, string(step), name)
+    mkpath(matrix_path)
+    if metadata != nothing
+        @assert length(metadata) == size(mat, 1) "#labels must be equal to #samples"
+        if metadata_header == nothing
+            metadata_header = [string(x) for x in metadata]
+        else
+            @assert length(metadata_header) == size(metadata, 2) "length of header must be equal to the number of columns in metadata"
+            metadata = vcat([join(metadata_header, '\t')], [join(x, '\t') for x in metadata])
+        end
+        write_metadata(metadata, matrix_path, metadata_header)
+    end
+    write_matrix(mat, matrix_path)
+    write_pbtext(name, logger.logdir, matrix_path, metadata, step)
+end
+
+function write_matrix(mat::AbstractArray, matrix_path::AbstractString)
+    matrix_path = joinpath(matrix_path, "tensor.tsv")
+    file = open(matrix_path, "w")
```

Comment:
Rather than using `open`-`close` pairs.
I suggest using:
```
open(matrix_path, "w") do file
...
end
```

which will automatically call `close` at th e end of the block.
and also handle closing in case of errors being thrown etc.